File: Python/remove_duplicates_linkedList.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def deleteDuplicates(self, head: Optional[ListNode]) -> Optional[ListNode]:
        curr = head
        while curr and curr.next is not None:
            val = curr.val
            if curr.next.val == val:
                curr.next = curr.next.next
            else:
                curr = curr.next
        return head

File: Cpp/remove_duplicates_linkedList.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* deleteDuplicates(ListNode* head) {
        ListNode* curr = head;
        while (curr != nullptr && curr -> next != nullptr)
        {
            int temp_val = curr -> val;
            ListNode* next_ptr =  curr -> next;
            int next_val = next_ptr -> val;
            if ( temp_val == next_val) curr -> next = next_ptr -> next;
            else curr =  curr -> next;
        }
        return head;
        
    }
};